[Bug] Windows CI is failing (again)

e.g. https://github.com/fzyzcjy/flutter_rust_bridge/runs/5312283556?check_suite_focus=true

There does not seem to be related code change that makes CI fail...

> LINK : fatal error LNK1104: cannot open file '..\Debug\flutter_rust_bridge_example.dll.lib' 
My guess is that something changed with MSVC build tools recently, we can try using the old `windows-2019` environment instead. Might be relevant to Corrosion as well.

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -180,7 +180,7 @@ jobs:
 
   flutter_windows_test:
     name: Flutter (Windows) integration test
-    runs-on: windows-latest
+    runs-on: windows-2019
 
     steps:
       - name: Install Rust toolchain

--- a/frb_codegen/src/others.rs
+++ b/frb_codegen/src/others.rs
@@ -13,12 +13,12 @@ use regex::RegexBuilder;
 // NOTE please sync [DUMMY_WIRE_CODE_FOR_BINDGEN] and [EXTRA_EXTERN_FUNC_NAMES]
 pub const DUMMY_WIRE_CODE_FOR_BINDGEN: &str = r#"
     // ----------- DUMMY CODE FOR BINDGEN ----------
-    
+
     // copied from: allo-isolate
     pub type DartPort = i64;
     pub type DartPostCObjectFnType = unsafe extern "C" fn(port_id: DartPort, message: *mut std::ffi::c_void) -> bool;
     #[no_mangle] pub unsafe extern "C" fn store_dart_post_cobject(ptr: DartPostCObjectFnType) { panic!("dummy code") }
-    
+
     // copied from: frb_rust::support.rs
     #[repr(C)]
     pub struct WireSyncReturnStruct {
@@ -26,7 +26,7 @@ pub const DUMMY_WIRE_CODE_FOR_BINDGEN: &str = r#"
         pub len: i32,
         pub success: bool,
     }
-    
+
     // ---------------------------------------------
     "#;
 
@@ -144,7 +144,7 @@ pub fn auto_add_mod_to_lib_core(rust_crate_dir: &str, rust_output_path: &str) ->
         .to_str()
         .ok_or_else(|| anyhow!(""))?
         .to_string()
-        .replace("/", "::");
+        .replace('/', "::");
     let expect_code = format!("mod {};", mod_name);
 
     let path_lib_rs = path_src_folder.join("lib.rs");

--- a/frb_example/pure_dart/rust/src/api.rs
+++ b/frb_example/pure_dart/rust/src/api.rs
@@ -272,7 +272,7 @@ pub struct ExoticOptionals {
 
 pub fn handle_optional_increment(opt: Option<ExoticOptionals>) -> Option<ExoticOptionals> {
     fn manipulate_list<T>(src: Option<Vec<T>>, push_value: T) -> Option<Vec<T>> {
-        let mut list = src.unwrap_or_else(Vec::new);
+        let mut list = src.unwrap_or_default();
         list.push(push_value);
         Some(list)
     }
@@ -289,7 +289,7 @@ pub fn handle_optional_increment(opt: Option<ExoticOptionals>) -> Option<ExoticO
         float32list: manipulate_list(opt.float32list, 0.),
         float64list: manipulate_list(opt.float64list, 0.),
         attributes: Some({
-            let mut list = opt.attributes.unwrap_or_else(Vec::new);
+            let mut list = opt.attributes.unwrap_or_default();
             list.push(Attribute {
                 key: "some-attrib".to_owned(),
                 value: "some-value".to_owned(),
@@ -297,7 +297,7 @@ pub fn handle_optional_increment(opt: Option<ExoticOptionals>) -> Option<ExoticO
             list
         }),
         nullable_attributes: Some({
-            let mut list = opt.nullable_attributes.unwrap_or_else(Vec::new);
+            let mut list = opt.nullable_attributes.unwrap_or_default();
             list.push(None);
             list
         }),

--- a/justfile
+++ b/justfile
@@ -51,10 +51,10 @@ clean:
     cd {{frb_flutter}}/rust && cargo clean
 
 refresh_all:
-    (cd frb_rust && cargo check)
-    (cd frb_macros && cargo check)
-    (cd frb_example/pure_dart/rust && cargo check)
-    (cd frb_example/with_flutter/rust && cargo check)
+    (cd frb_rust && cargo clippy -- -D warnings)
+    (cd frb_macros && cargo clippy -- -D warnings)
+    (cd frb_example/pure_dart/rust && cargo clippy -- -D warnings)
+    (cd frb_example/with_flutter/rust && cargo clippy -- -D warnings)
     (cd frb_example/pure_dart/dart && dart pub get)
     (cd frb_example/with_flutter && flutter pub get)
 

